inlong-sort/sort-connectors/doris/src/main/java/org/apache/inlong/sort/doris/table/DorisDynamicSchemaOutputFormat.java

```diff
@@ -258,35 +265,52 @@ public synchronized void close() throws IOException {
 
     @SuppressWarnings({"rawtypes"})
     public synchronized void flush() {
+        flushing = true;
```

Comment:
volatile can't ensure CAS atomic operation?

Reply:
The entire flush needs to ensure atomicity here.